bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/internal/NetworkUtils.java

```diff
@@ -0,0 +1,153 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.broadlink.internal;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.ServerSocket;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.Enumeration;
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.broadlink.internal.handler.BroadlinkHostNotReachableException;
+import org.slf4j.Logger;
+
+/**
+ * Utilities for working with the local network.
+ *
+ * @author John Marshall/Cato Sognen - Initial contribution
+ */
+@NonNullByDefault
+public class NetworkUtils {
+
+    /**
+     * Check whether the host is available.
+     *
+     * @param host hostname or ip address of the host to check for availability
+     * @param timeout time in milliseconds for the check to timeout
+     * @param logger the logger to use for logging any issues
+     * @throws IOException if there is an unexpected error
+     * @throws BroadlinkHostNotReachableException when the device is not responding
+     */
+
+    public static void hostAvailabilityCheck(@Nullable String host, int timeout, Logger logger)
+            throws IOException, BroadlinkHostNotReachableException {
+        InetAddress address = InetAddress.getByName(host);
+        if (!address.isReachable(timeout)) {
+            throw new BroadlinkHostNotReachableException("Cannot reach " + host);
+        }
+    }
+
+    /**
+     * Finds an InetAddress that is associated to a non-loopback device.
+     *
+     * @return null, if there is no non-loopback device address, otherwise the InetAddress associated to a non-loopback
+     *         device
+     * @throws SocketException thrown when no socket can be opened.
+     */
+    public static @Nullable InetAddress findNonLoopbackAddress() throws SocketException {
+        Enumeration<NetworkInterface> ifaces = NetworkInterface.getNetworkInterfaces();
+        while (ifaces.hasMoreElements()) {
+            NetworkInterface iface = ifaces.nextElement();
+            Enumeration<InetAddress> inetAddrs = iface.getInetAddresses();
+            while (inetAddrs.hasMoreElements()) {
+                InetAddress inetAddr = inetAddrs.nextElement();
+                if (inetAddr.isLoopbackAddress()) {
+                    continue; /* Loop/switch isn't completed */
+                }
+
+                if (inetAddr.isSiteLocalAddress()) {
+                    return inetAddr;
+                }
+            }
+        }
+
+        return null;
```

Comment:
Simplify by making use of the framework available API's

```suggestion
        for (InetAddress address : NetUtil.getAllInterfaceAddresses().stream().filter(a->a.getAddress() instanceof Inet4Address).map(a->a.getAddress()).toList()) {
            if (address.isSiteLocalAddress()) {
                return address;
            }
        }
        return null;
```